mongo/integration/unified/server_api_options_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package unified
+
+import (
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// ServerAPIOptions is a wrapper for *options.ServerAPIOptions. This type implements the bson.Unmarshaler interface
+// to convert BSON documents to a ServerAPIOptions instance.
+type ServerAPIOptions struct {
+	*options.ServerAPIOptions
+}
+
+type ServerAPIVersion = options.ServerAPIVersion
+
+var _ bson.Unmarshaler = (*ServerAPIOptions)(nil)
+
+func (to *ServerAPIOptions) UnmarshalBSON(data []byte) error {
+	var temp struct {
+		ServerAPIVersion  ServerAPIVersion       `bson:"version"`
+		DeprecationErrors bool                   `bson:"deprecationErrors"`
+		Strict            bool                   `bson:"strict"`
+		Extra             map[string]interface{} `bson:",inline"`
+	}
+	if err := bson.Unmarshal(data, &temp); err != nil {
+		return fmt.Errorf("error unmarshalling to temporary ServerAPIOptions object: %v", err)
+	}
+	if len(temp.Extra) > 0 {
+		return fmt.Errorf("unrecognized fields for ServerAPIOptions: %v", MapKeys(temp.Extra))
+	}
+
+	to.ServerAPIOptions = options.ServerAPI()
+	if err := temp.ServerAPIVersion.CheckServerAPIVersion(); err != nil {
```

Comment:
We probably don't want to do this validation here. Otherwise, we're not testing how a `mongo.Client` would behave if an invalid server API version were specified. Can we remove the call here and let client construction error later on?

Reply:
Removed it for now. I had it there because of this line in the updated unified-test-format spec: `A ``version`` string is required, and test runners MUST fail if the given version string is not supported by the driver`. It said test runners and not tests, but you are right that we wouldn't be testing client behavior for invalid server API versions...

Reply:
Discussed in Slack. The spec only mandates that this validation be done for drivers that don't have support for enums, so there wouldn't be a spec test for this. Let's add back this check here and when you actually add the code to call `ServerAPIVersion.Validate()` during client configuration, you can add an integration test to assert that we error on invalid values.

Reply:
Added back in! And that sounds good, will make a note about the integration test.